src/helpers/dateTimeHelper.ts

```diff
@@ -0,0 +1,28 @@
+import moment from 'moment';
+import { momentFormat } from './momentHelper';
+
+/**
+ * Format event date strings with momentjs to a beautiful human readable string.
+ *
+ * @param dateStart the start date of format 'YYYY-MM-DD'
+ * @param dateEnd the end date of format 'YYYY-MM-DD'
+ *
+ * @return the formated event date string in format 'DD.MM.YYYY'
+ */
+export const eventDate = (dateStart?: string, dateEnd?: string) => {
+  if (!dateStart && !dateEnd) {
+    return '';
+  } else {
+    // by the previous check one of the two needs to be defined for moment format
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    return momentFormat((dateStart || dateEnd)!);
```

Comment:
what does the `!` at the end? 
how does the comment fit when writing `by the previous check one of the two needs to be defined`? the previous check is the if statement? moment is called for both values existing, right? i do not get it completely here!?

why have you added an explicit `else` here?

Reply:
the `!` tells typescript that that value is not undefined (it is usually better to avoid that, because it introduces possible loss of typeSafety.

the `momentFormat` is called if EITHER of them is undefined, which causes the disjunction to be always result in a defined value. however typescripts inference does not catch that at least one of them has to be defined if the code is written this way.

i changed it now, to make it clearer.